### ISSUE TITLE
Improve the check for whether _heisenberg_rep() is implemented

### ISSUE THREAD

**Issue by cgogolin** [XanaduAI/pennylane#154]
Having to construct parameters to be able to call self._heisenberg_rep() and then actually construcitng the rep is not a very elegant way of checking whether a heisenberg rep is in principle supported. Can one not do something like `Operation._heisenberg_rep == type(self)._heisenberg_rep` to test whether the method was overwritten?

This litte example suggests this should be possible:
```
class A:
    def m():
        return 'a'

    def check_if_m_overwritten(self):
        return A.m == type(self).m

class B(A):
    pass

class C(A):
    def m():
        return 'b'


b = B()
c = C()

print(b.check_if_m_overwritten())
print(c.check_if_m_overwritten())
```
Output:
```
True
False
```

_Originally posted by @cgogolin in https://github.com/XanaduAI/pennylane/pull/149_

**Comment by cgogolin** [comment]
I will do that tomorrow

**Comment by cgogolin** [comment]
fixed in #155
